### ISSUE TITLE
Plugin enhance

### ISSUE THREAD

**Issue by belongwqz** [apache/apisix#3393]
Currently the plugin `response-rewrite` support diy body or encode_base64 body, i's still relatively weak, if it can modify the body, it should be better. For example:
before body:
`{"key": "value"}`
after body:
if the status code<400 `{"status": 0, "msg": "OK", "data": {{ xxx-response_body }}`
if the status code>=400 `{"status": {{ status_code }}, "msg": {{ xxx-response_body }}}`
Maybe it can use regular expression to support this.

**Comment by tokers** [comment]
@belongwqz Actually you can use `serverless` plugin to customize your demand.

**Comment by belongwqz** [comment]
> @belongwqz Actually you can use `serverless` plugin to customize your demand.

All right, maybe we can give more examples in the `serverless` plugin documentation to show it's power.

**Comment by tokers** [comment]
> > @belongwqz Actually you can use `serverless` plugin to customize your demand.
> 
> 
> 
> All right, maybe we can give more examples in the `serverless` plugin documentation to show it's power.

Yeah! That's more like best practices about this plugin, welcome to submit pr!
